editor/error-boundary/index.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button, ClipboardButton } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import Warning from '../warning';
+
+class ErrorBoundary extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.reboot = this.reboot.bind( this );
+		this.getStateText = this.getStateText.bind( this );
+
+		this.state = {
+			hasEncounteredError: false,
+		};
+	}
+
+	componentDidCatch() {
+		this.setState( { hasEncounteredError: true } );
+	}
+
+	reboot() {
+		this.props.onError( this.context.store.getState() );
+	}
+
+	getStateText() {
+		return JSON.stringify( this.context.store.getState() );
+	}
+
+	render() {
+		const { hasEncounteredError } = this.state;
+		if ( ! hasEncounteredError ) {
+			return this.props.children;
+		}
+
+		return (
+			<Warning>
+				<p>{ __(
+					'The editor has suffered from a catastrophic error.'
+				) }</p>
+				<p>
+					<Button onClick={ this.reboot } isLarge>
+						{ __( 'Attempt Recovery' ) }
+					</Button>
+					<ClipboardButton text={ this.getStateText } isLarge>
```

Comment:
In addition to the state, is there a way to provide the stack trace. People could send us both to debug quickly.

Reply:
Yeah, that makes sense. Maybe they can be combined into a single "Copy Debug Info" button?

Reply:
Sounds good to me